metadata_backend/helpers/metax_api_handler.py

```diff
@@ -0,0 +1,261 @@
+                # default
+                "access_rights": {
+                    "access_type": {"identifier": "http://uri.suomi.fi/codelist/fairdata/access_type/code/restricted"},
+                },
+                # default
+                "publisher": {
+                    "@type": "Organization",
+                    "name": {
+                        "en": "CSC Sensitive Data Services for Research",
+                        "fi": "CSC:n Arkaluonteisen datan palveluiden aineistokatalogi",
+                    },
+                },
+            },
+        }
+
+    async def get_metadata_provider_user(self) -> str:
+        """Get current user's external id.
+
+        returns: current users external ID
+        """
+        current_user = get_session(self.req)["user_info"]
+        user_op = UserOperator(self.db_client)
+        user = await user_op.read_user(current_user)
+        metadata_provider_user = user["externalId"]
+        return metadata_provider_user
+
+    async def post_dataset_as_draft(self, collection: str, data: Dict) -> str:
+        """Send draft dataset to Metax.
+
+        Construct Metax dataset data from submitters' Study or Dataset and
+        send it as new draft dataset to Metax Dataset API.
+
+        :param collection: schema of incomming submitters metadata
+        :param data: validated Study or Dataset data dict
+        :raises: HTTPError depending on returned error from Metax
+        :returns: Metax ID for dataset returned by Metax API
+        """
+        metax_dataset = self.minimal_dataset_template
+        metax_dataset["metadata_provider_user"] = await self.get_metadata_provider_user()
+        if collection == "dataset":
+            dataset_data = await self.create_metax_dataset_data_from_dataset(data)
+        else:
+            dataset_data = await self.create_metax_dataset_data_from_study(data)
+        metax_dataset["research_dataset"] = dataset_data
+        LOG.debug(
+            f"Creating draft dataset to Metax service from Submitter {collection} with accession ID "
+            f"{data['accessionId']}"
+        )
+        async with aiohttp.ClientSession() as sess:
+            resp = await sess.post(
+                f"{self.metax_url}{self.rest_route}",
+                params="draft",
+                json=metax_dataset,
+                auth=aiohttp.BasicAuth(self.username, self.password),
+            )
+            status = resp.status
+            if status == 201:
+                metax_data = await resp.json()
+                LOG.debug(
+                    f"Created Metax draft dataset {metax_data['identifier']} from Submitter {collection} "
+                    f"{data['accessionId']} with data: {metax_dataset}."
+                )
+                return metax_data["identifier"]
+            else:
+                # TODO: how front end should react on this??
+                reason = await resp.text()
+                raise self.process_error(status, reason)
```

Comment:
@hannyle @saulipurhonen

Reply:
this is already handled in the middleware with the redirect to the error pages, not sure why we need to do more here than raising the error. See: https://github.com/CSCfi/metadata-submitter/blob/develop/metadata_backend/api/middlewares.py#L18-L71